fabric-convention-tags-v2/src/main/java/net/fabricmc/fabric/api/tag/convention/v2/TagUtil.java

```diff
@@ -81,4 +82,29 @@ public static <T> boolean isIn(@Nullable DynamicRegistryManager registryManager,
 
 		return false;
 	}
+
+	/**
+	 * Use this to get a TagKey's translation key safely on any side.
+	 * @return the translation key for a TagKey.
+	 */
+	public static String getTagTranslationKey(TagKey<?> tagKey) {
```

Comment:
I am going to assume that the `isIn` method in TagUtil isn't injected into TagKey because it could lead to discoverability issues. Fabric doesn't do patches like forge afaik so people looking in TagKey won't see the inject method as it'll only be applied by mixin.

Reply:
There's nothing stopping an intjected interface+mixin for these.

TagUtil is separate as it didn't need to be on the tagkey, plus `isIn` does not make sense to have on the tagkey and would need to be inverted, eg. `myTag.contains(Blocks.DIRT)`. It was separate to help with discovery of the method, as other tag queries are done on the object, not the tag. [See the PR.](https://github.com/FabricMC/fabric/pull/2307)

Reply:
Is there an example of an interface injection in Fabric API that I can look at for how it is setup?

Reply:
Nvm, someone got me an example to checkout. I'll give it a shot
https://github.com/FabricMC/fabric/blob/109a65e5af40e56b017f8db1bda6282a8ded20e8/fabric-block-view-api-v2/src/main/resources/fabric.mod.json#L31-L34

Reply:
Just to clarify, moving the translation to an injected interface is good as it's a property of the tagkey, but should not be done for `isIn`.